common/beerocks/hostapd/include/configuration.h

```diff
@@ -123,6 +124,16 @@ class Configuration {
      */
     template <class func> void for_all_ap_vaps(func);
 
+    /**
+     * @brief apply func to all ap vaps, increment the given iterator 
+     * with each call to func: f(current_vap, ++user_itertor)
+     * func interface: template<class iter> f(const std::string& vap, iter user_iterator)
+     * @details apply func to all vaps that thier mode is "ap" while incrementing
+     * the user iterator. this functionality enables iterating over two containers
+     * in paralel: the ap-vaps and the user container.
+     */
+    template <class func, class iter> void for_all_ap_vaps(func, iter current, const iter end);
+
```

Comment:
What we *actually* need is the possibility to iterate over all vaps in parallel with the function, and then perform another function on the remaining vaps.